### ISSUE TITLE
Triangle example fails on current master

### ISSUE THREAD

**Issue by porky11** [gfx-rs/wgpu-native#56]
I just run tihs:
```
make run-triangle-example
```

Exact commit is `f8b861555893e2bbe6011958cb1c8d92af3974bc`.

Here's the backtrace:
```
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', /home/porky11/.cargo/git/checkouts/wgpu-53e70f8674b08dd4/8ce2530/wgpu-core/src/hub.rs:755:33
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1076
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1537
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:217
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:526
  11: rust_begin_unwind
             at src/libstd/panicking.rs:437
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
  13: core::panicking::panic
             at src/libcore/panicking.rs:50
  14: core::option::Option<T>::unwrap
             at /mnt/data/home/porky11/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/macros/mod.rs:10
  15: <gfx_backend_vulkan::Backend as wgpu_core::hub::GfxBackend>::get_surface_mut
             at /home/porky11/.cargo/git/checkouts/wgpu-53e70f8674b08dd4/8ce2530/wgpu-core/src/hub.rs:755
  16: wgpu_core::device::<impl wgpu_core::hub::Global<G>>::device_create_swap_chain
             at /home/porky11/.cargo/git/checkouts/wgpu-53e70f8674b08dd4/8ce2530/wgpu-core/src/device/mod.rs:3525
  17: wgpu_device_create_swap_chain
             at src/device.rs:890
  18: main
             at ./main.c:197
  19: __libc_start_main
             at ../csu/libc-start.c:308
  20: _start
             at ../sysdeps/x86_64/start.S:120
```

The compute example works fine.

**Comment by DevOrc** [comment]
I'm getting the same issue on Ubuntu but it works fine on Windows 10

**Comment by kvark** [comment]
Should be fixed by c75e5c2c427ef0e1010551235c50e22c2f6a606d
